$LAAS: pipo 2008/09/15 17:36:57 mallet $

Fix detection of tcl on x86_64 for generated templates

--- canvas/autoconf/robots.m4	2008-08-29 15:03:39.000000000 +0200
+++ canvas/autoconf/robots.m4~	2008-09-13 13:26:35.000000000 +0200
@@ -327,20 +343,31 @@
 
 
 dnl --- Look for tcl ----------------------------------------------------
-dnl ROBOT_LIB_TCL EXTRA_DIRECTORY
+dnl
 AC_DEFUN([ROBOT_LIB_TCL],
 [
+   case $host_cpu-$host_os in
+      x86_64-linux*)
+	libsuffix=64
+	;;
+      *)
+	libsuffix=
+	;;
+   esac
+
    AC_ARG_WITH(tcl,
-      AC_HELP_STRING([--with-tcl=DIR], [directory containing tclConfig.sh]),
+      AC_HELP_STRING([--with-tcl=DIR], [directory containing tclConfig.sh. 'no' to disable.]),
       [tcl_prefix=$withval],
       [for ac_dir in \
-         $1 \
-         ${exec_prefix}/lib      \
-         /usr/local/lib/tcl8.4   \
-         /usr/local/lib          \
-         /usr/pkg/lib            \
-         /usr/lib/tcl8.4         \
-         /usr/lib                \
+         ${exec_prefix}/lib      	\
+         /usr/local/lib/tcl8.4   	\
+         /usr/local/lib/tcl8.3   	\
+         /usr/local/lib          	\
+         /usr/pkg/lib            	\
+         /usr/lib${libsuffix} 		\
+         /usr/lib${libsuffix}/tcl8.4	\
+         /usr/lib${libsuffix}/tcl-8.4	\
+         /usr/lib${libsuffix}/tcl8.3	\
         ; \
        do
          if test -r "$ac_dir/tclConfig.sh"; then
@@ -349,16 +376,21 @@
          fi
        done])
 
-   AC_MSG_CHECKING([for tclConfig.sh])
-   if test -r "${tcl_prefix}/tclConfig.sh"; then
-      file=${tcl_prefix}/tclConfig.sh
-      . $file
-      AC_MSG_RESULT("${tcl_prefix}/tclConfig.sh")
+   if test "x${tcl_prefix}" = "xno"; then
+	HAS_TCL=no
    else
-      AC_MSG_RESULT([not found (fatal)])
+
+   AC_MSG_CHECKING([for tclConfig.sh])
+   if test ! -r "${tcl_prefix}/tclConfig.sh"; then
+      AC_MSG_RESULT([not found: DISABLING tclServ compilation])
       AC_MSG_RESULT([Please use --with-tcl to specify a valid path to your tclConfig.sh file]) 
-      exit 2;
-   fi
+   else
+
+   TCL_CONFIG_PATH=${tcl_prefix}
+   file=${tcl_prefix}/tclConfig.sh
+   . $file
+   AC_MSG_RESULT("${tcl_prefix}/tclConfig.sh")
+
    dnl substitute variables in TCL_LIB_FILE
    eval TCL_LIB_FILE=${TCL_LIB_FILE}
 
@@ -379,12 +411,17 @@
          break
       fi
    done
-   if test "x$ac_tcl_includes" != "x"; then
-      AC_MSG_RESULT($ac_tcl_includes)
+   if test "x$ac_tcl_includes" = "x"; then
+      echo
+      echo "+--------------------------------------------------+"
+      echo "|  If tclConfig.sh was found under /usr/lib/ but   |"
+      echo "|  the Tcl headers could not be found, you should  |"
+      echo "|  use --with-tcl=/path/to/tcl/headers             |"
+      echo "+--------------------------------------------------+"
+      AC_MSG_ERROR([not found (fatal)])
    else
-      AC_MSG_RESULT([not found (fatal)])
-      exit 2;
-   fi
+
+   AC_MSG_RESULT($ac_tcl_includes)
 
    AC_MSG_CHECKING([for tcl library])
    test -z "$tcl_test_lib" && tcl_test_lib="${TCL_LIB_FILE}"
@@ -392,7 +429,7 @@
       $TCL_EXEC_PREFIX/lib                    \
       $TCL_PREFIX/lib                         \
       /usr/local/lib                          \
-      /usr/lib                                \
+      /usr/lib${libsuffix}                    \
       /Library/Frameworks/Tcl.framework       \
       $extra_lib                              \
       ; \
@@ -402,12 +439,11 @@
          break
       fi
    done
-   if test "x$ac_tcl_libs" != "x"; then
-      AC_MSG_RESULT($ac_tcl_libs/$TCL_LIB_FILE)
-   else
+   if test "x$ac_tcl_libs" = "x"; then
       AC_MSG_RESULT([not found (fatal)])
-      exit 2;
-   fi
+   else
+
+   AC_MSG_RESULT($ac_tcl_libs/$TCL_LIB_FILE)
 
    if test "$ac_tcl_includes" != "/usr/include"; then
       TCL_CPPFLAGS="-I$ac_tcl_includes"
@@ -419,6 +455,16 @@
    else 
       TCL_LDFLAGS=""
    fi
+
+   HAS_TCL=yes
+
+   fi # libtcl not found
+   fi # tcl.h not found
+   fi # tclConfig.sh not found
+   fi # --with-tcl=no
+
+   AC_SUBST(HAS_TCL)
+   AC_SUBST(TCL_CONFIG_PATH)
    AC_SUBST(TCL_CPPFLAGS)
    AC_SUBST(TCL_LDFLAGS)
    AC_SUBST(TCL_LIBS)
